pkg/skaffold/runner/dev.go

```diff
@@ -42,43 +61,40 @@ func (r *SkaffoldRunner) Dev(ctx context.Context, out io.Writer, artifacts []*la
 	defer forwarderManager.Stop()
 
 	// Create watcher and register artifacts to build current state of files.
-	changed := changes{}
```

Comment:
I liked when this state was local to this function. This way, it was clear that nobody else could change its state.

Reply:
`changeSet` is a better name though.

Reply:
I see your point, it's much easier to handle the tests when the `changeSet` is embedded on the runner though. plus, it's not really true that "nobody can change its state": the file monitor does exactly this, albeit through the callbacks that we register before we start the dev loop.

if you feel strongly about it, I can change it back.